src/hearings/containers/request-hearing/hearing-link/hearing-link.component.ts

```diff
@@ -48,25 +49,30 @@ export class HearingLinkComponent extends RequestHearingPageFlow implements OnIn
   }
 
   public generateLinkedCasesWithReasonDescription(): void {
-    this.hearingsServiceSub = this.hearingsService.loadCaseLinkingReasonCodes().subscribe(reasons => {
-      this.hearingLinksSub = this.hearingStore.pipe(select(fromHearingStore.getHearingLinks)).subscribe(
-        hearingLinks => {
-          if (hearingLinks.serviceLinkedCases) {
-            hearingLinks.serviceLinkedCases.forEach(linkedCase => {
-              const caseLinkingReasons = reasons.list_of_values.filter(reason => linkedCase.reasonsForLink.some(reasonCode => reason.key === reasonCode));
-              const caseLinkingReasonsValues = caseLinkingReasons.map(x => x.value_en);
-              if (caseLinkingReasonsValues && caseLinkingReasonsValues.length > 0) {
-                this.linkedCases.push({caseName: linkedCase.caseName, caseReference: linkedCase.caseReference, reasonsForLink: caseLinkingReasonsValues});
-              } else {
-                this.linkedCases.push(linkedCase);
-              }
-            });
-            this.showSpinner = false;
-          }
+    this.hearingLinksSub = this.hearingsService.loadCaseLinkingReasonCodes().pipe(
+      switchMap(reasons => {
+        this.caseLinkingReasons = reasons;
+        return this.hearingStore.pipe(select(fromHearingStore.getHearingLinks))
+      })
+    ).subscribe({
+      next: (hearingLinks: HearingLinksStateData) => {
+        if (hearingLinks.serviceLinkedCases) {
+          this.linkedCases = [];
+          hearingLinks.serviceLinkedCases.forEach(linkedCase => {
+            const caseLinkingReasons = this.caseLinkingReasons.list_of_values.filter(reason => linkedCase.reasonsForLink.some(reasonCode => reason.key === reasonCode));
+            const caseLinkingReasonsValues = caseLinkingReasons.map(x => x.value_en);
+            if (caseLinkingReasonsValues && caseLinkingReasonsValues.length > 0) {
+              this.linkedCases.push({caseName: linkedCase.caseName, caseReference: linkedCase.caseReference, reasonsForLink: caseLinkingReasonsValues});
+            } else {
+              this.linkedCases.push(linkedCase);
+            }
+          });
         }
-      );
-    }, () => {
-      this.router.navigate(['/hearings/error']);
+        this.showSpinner = false;
+      },
+      error: () => {
+        this.router.navigate(['/hearings/error']);
+      }
```

Comment:
can you just remove this unnecessary wrapping in brackets - 

`error: () => this.router.navigate(['/hearings/error'])`

Reply:
I have done the changes as you suggested in this PR and in PR 2638.